### ISSUE TITLE
Leader GS node - online to offline and back, issue -

### ISSUE THREAD

**Issue by Saminobi** [kc1212/virtual-grid#27]
When previous leader GS revives while jobs are being processed, either that node shows an endless stream of messages or comes back online and all the other GS nodes crush.

### Scenario 1
- 5 GS
- 20 RM
- 10.000 jobs (x3)

1) The system runs smoothly, processing the job requests
2) The leader GS node (call it "x")  is shut down manually
3) The GS with the second highest id (call it "y") becomes the new leader and continues processing the job requests
4) Manually bring back online the previous GS leader ("x")
5) The job requests are still being processed by the new leader ("y") while the old leader ("x") is displaying a stream of messages

### Scenario 2
- 5 GS
- 4 RM
- 1.000 jobs

1) The system runs smoothly, processing the job requests
2) The leader GS node (call it "x")  is shut down manually
3) The GS with the second highest id (call it "y") becomes the new leader and continues processing the job requests
4) Manually bring back online the previous GS leader ("x")
5) The old leader ("x") displays a shorter stream of messages and when it acts normal, all the other GS nodes crush

##### Error Message of GS nodes:
```
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
panic(0x7fca20, 0xc82000e140)
        /usr/lib/go/src/runtime/panic.go:464 +0x3e6
github.com/kc1212/virtual-grid/model.(*GridSdr).scheduleJobs(0xc820130000)
        /home/jeisp/Go/src/github.com/kc1212/virtual-grid/model/gridsdr.go:180 +0x709
github.com/kc1212/virtual-grid/model.(*GridSdr).Run(0xc820130000)
        /home/jeisp/Go/src/github.com/kc1212/virtual-grid/model/gridsdr.go:105 +0x3f5
main.main()
        /home/jeisp/Go/src/github.com/kc1212/virtual-grid/cmd/gridsdr/main.go:20 +0x201
```

**Comment by kc1212** [comment]
Regarding scenario 2, how many worker nodes did you use, and you added 1000 jobs only once and not three times?

**Comment by kc1212** [comment]
First attempt to fix it on 5c19b2c

**Comment by Saminobi** [comment]
After last commit, the bug seems to have been fixed.
